source/common/router/router_ratelimit.cc

```diff
@@ -38,10 +38,11 @@ bool RequestHeadersAction::populateDescriptor(const Router::RouteEntry&,
                                               const Http::HeaderMap& headers,
                                               const Network::Address::Instance&) const {
   const Http::HeaderEntry* header_value = headers.get(header_name_);
+
+  // If header is not present and skip_if_absent_ is not set to true, short circuit here and return.
```

Comment:
nit: this comment is not quite accurate, as you return no matter what. Can you clarify?

Reply:
Hello @mattklein123 - Yes, you are right, I agree it can be misleading. Let me rephrase it. What I was going for was "do not add the descriptor".  Thanks for pointing it out!